src/Illuminate/Foundation/Exceptions/Handler.php

```diff
@@ -220,12 +218,18 @@ protected function prepareJsonResponse($request, Exception $e)
 
         $headers = $this->isHttpException($e) ? $e->getHeaders() : [];
 
-        return response(json_encode([
-            'message' => $e->getMessage(),
-            'file' => $e->getFile(),
-            'line' => $e->getLine(),
-            'trace' => $e->getTrace(),
-        ], JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES), $status, array_merge($headers, [
+        if (config('app.debug')) {
+            $response = [
+                'message' => $e->getMessage(),
+                'file' => $e->getFile(),
+                'line' => $e->getLine(),
+                'trace' => $e->getTrace(),
+            ];
+        } else {
+            $response['message'] = $this->isHttpException($e) ? $e->getMessage() : 'Server Error';
```

Comment:
I believe this to be clearer:

```php
$response = [
    'message' => $this->isHttpException($e) ? $e->getMessage() : 'Server Error',
];
```

...since we're only now creating the response.